client/contracts/truffle/test/oracle/OracleMgr.test.js

```diff
@@ -9,49 +9,188 @@ require('chai')
+      this.consensus = await ConsensusMock.new(true);
+      await this.resolver.setAddr(namehash('validator.kowala'), this.consensus.address, { from: owner });
+      this.oracle = await OracleMgr.new(1, 1, 1, this.resolver.address, { from: owner });
+    });
+
+    it('should set Consensus address using KNS', async () => {
+      // given
+      const knsResolverAddr = await this.oracle.knsResolver();
+      const resolver = await PublicResolver.at(knsResolverAddr);
+
+      // when
+      const consensusAddr = await resolver.addr(namehash('validator.kowala'));
+
+      // then
+      await consensusAddr.should.be.equal(this.consensus.address);
+    });
   });
 
-  it('should set Consensus address using KNS', async () => {
-    // given
-    const knsResolverAddr = await this.oracle.knsResolver();
-    const resolver = await PublicResolver.at(knsResolverAddr);
+  describe('Functionality of', async () => {
+    beforeEach(async () => {
+      this.consensus = await ConsensusMock.new(true);
+      this.resolver = await DomainResolverMock.new(this.consensus.address);
+      this.oracle = await OracleMgr.new(3, 1, 1, this.resolver.address, { from: owner });
+    });
+    describe('registration', async () => {
+      it('should register oracle', async () => {
+        // when
+        await this.oracle.registerOracle({ from: newOwner });
+
+        // then
+        const oracleCount = await this.oracle.getOracleCount();
+        await oracleCount.should.be.bignumber.equal(1);
+      });
+
+      it('should not register already registered oracle', async () => {
+        // given
+        await this.oracle.registerOracle({ from: newOwner });
+
+        // when
+        const expectedRegistrationFailure = this.oracle.registerOracle({ from: newOwner });
+
+        // then
+        await expectedRegistrationFailure.should.eventually.be.rejectedWith(EVMError('revert'));
+      });
+
+      it('should not register oracle above oracle limit', async () => {
+        // given
+        await this.oracle.registerOracle({ from: newOwner });
+        await this.oracle.registerOracle({ from: newOwner2 });
+        await this.oracle.registerOracle({ from: newOwner3 });
+
+        // when
+        const expectedRegistrationFailure = this.oracle.registerOracle({ from: newOwner4 });
+
+        // then
+        await expectedRegistrationFailure.should.eventually.be.rejectedWith(EVMError('revert'));
+      });
+    });
+
+    describe('deregistration', async () => {
+      it('should deregister oracle', async () => {
+        // given
+        await this.oracle.registerOracle({ from: newOwner });
+
+        // when
+        await this.oracle.deregisterOracle({ from: newOwner });
+
+        // then
+        const oracleCount = await this.oracle.getOracleCount();
+        await oracleCount.should.be.bignumber.equal(0);
+      });
+
+      it('should not deregister deregistered oracle', async () => {
```

Comment:
not sure which scenario is this one - I think that one below is enough

Reply:
Trying deregister, already deregistered oracle. This should not be possible. That's what I'm testing